src/diffusers/models/attention_processor.py

```diff
@@ -1234,11 +1239,16 @@ def __call__(
         inner_dim = key.shape[-1]
         head_dim = inner_dim // attn.heads
 
-        query = query.view(batch_size, -1, attn.heads, head_dim).transpose(1, 2)
-
-        key = key.view(batch_size, -1, attn.heads, head_dim).transpose(1, 2)
-        value = value.view(batch_size, -1, attn.heads, head_dim).transpose(1, 2)
-
+        if attn.attention_legacy_order:
+            qkv = torch.cat([query, key, value], dim=2).transpose(1, 2)
+            query, key, value = qkv.reshape(batch_size, attn.heads, head_dim * 3, -1).chunk(3, dim=2)
+            query = query.transpose(-1, -2)
+            key = key.transpose(-1, -2)
+            value = value.transpose(-1, -2)
```

Comment:
We should be able to get rid of this if we incorporate this during the conversion. See how DiT does it:

https://github.com/huggingface/diffusers/blob/acd1962769715cebee1bb8e87075db4e78490070/scripts/convert_dit_to_diffusers.py#L58